core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -118,6 +118,17 @@
     public static final Setting<Integer> MAX_TOKEN_COUNT_SETTING =
         Setting.intSetting("index.analyze.max_token_count", 10000, 1, Property.Dynamic, Property.IndexScope);
 
+
+    /**
+     * A setting describing the maximum number of characters that will be analyzed for a highlight request.
+     * This setting is only applicable when highlighting is requested on a text that was indexed without
+     * offsets or term vectors.
+     * The default maximum of 10000 characters is defensive as for highlighting larger texts,
+     * indexing with offsets or term vectors is recommended.
+     */
+    public static final Setting<Integer> MAX_ANALYZED_OFFSET_SETTING =
+        Setting.intSetting("index.analyze.max_analyzed_offset", 10000, 1, Property.Dynamic, Property.IndexScope);
```

Comment:
Maybe `index.highlight.max_analyzed_offset` to indicate that it applies to highlighting only ?